rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+#![feature(once_cell)]
+
+pub mod checker {
+    use crate::utils::{c_char_to_vec, vec_to_c_char};
+    use libc::c_char;
+    use prover::zkevm::{CircuitCapacityChecker, RowUsage};
+    use serde_derive::{Deserialize, Serialize};
+    use std::cell::OnceCell;
+    use std::collections::HashMap;
+    use std::panic;
+    use std::ptr::null;
+    use types::eth::BlockTrace;
+
+    #[derive(Debug, Clone, Deserialize, Serialize)]
+    pub struct RowUsageResult {
+        pub acc_row_usage: Option<RowUsage>,
+        pub tx_row_usage: Option<RowUsage>,
+        pub error: Option<String>,
+    }
+
+    static mut CHECKERS: OnceCell<HashMap<u64, CircuitCapacityChecker>> = OnceCell::new();
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn init() {
+        env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("debug"))
+            .format_timestamp_millis()
+            .init();
+        let checkers = HashMap::new();
+        CHECKERS.set(checkers).unwrap();
```

Comment:
```suggestion
        CHECKERS.set(checkers).expect("circuit capacity checker initialized twice");
```

I'd suggest using `expect` at other places in this file too for easier debugging.

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/321/commits/714afd721af62aa19b2725994f8154509950f629